test/abstractarray.jl

```diff
@@ -1280,3 +1280,33 @@ end
     @test Int[0 t...; t... 0] == [0 1 2; 1 2 0]
     @test_throws ArgumentError Int[t...; 3 4 5]
 end
+
+@testset "issue #39896, modified getindex " begin
+    #Testing types holds for the range types in Base that use 1-based indexing
+    a=collect(1:10)
+    ax=axes(a,1)
+    @test a==a[ax]
+    @test typeof(a)===typeof(a[ax])
+    a=reshape(collect(1:100),(10,10))
+    @test a[2:5]==2:5
+    a=collect(1:10)
+    ax=Base.OneTo(8)
+    @test a[1:8]==a[ax]
+    @test typeof(a[1:8])===typeof(a[ax])
+    a=reshape(collect(1:100),(10,10))
+    @test a[1:8,1]==a[ax,1]
+    @test typeof(a[1:8,1])===typeof(a[ax,1])
```

Comment:
These should be changed to linear indexing as well.

Reply:
`@test typeof(arr[inds]) === typeof(arr)` shall fail as typeof(arr) will be Array{Type,2} for 2-d array and Array{Type,3} for 3-d array, while typeof(arr[inds]) will be Array{Type,1}. Instead should i use eltype in place of typeof or use typeof like as follows
`@test typeof(arr[inds]) === typeof(arr[:])`

Reply:
I think testing for `eltype` and `ndims` separately should work. Testing for the exact type might be too strict, and somewhat unnecessary. The `eltype` should be conserved and `ndims` should be 1.

Reply:
You can also do `@test arr[inds] isa Vector{eltype(arr)}` (or `@test arr[inds] isa AbstractVector{eltype(arr)}` if `Vector` is too strict).

Reply:
Changes added as asked.